core/cap-0046-06.md

```diff
@@ -354,6 +401,60 @@ differently even though they have the same format.
 Uniqueness is not a concern for smart tokens, so their identifiers are
 constructed using the typical creator and salt approach.
 
+### Clawback
+
+The `clawback` function is enabled by default for non-classic tokens. For
+classic tokens, `clawback` will be enabled only if `AUTH_CLAWBACK_ENABLED_FLAG`
+is set on the issuer. If the issuer account is missing, `clawback` will be
+disabled on initialization because anyone could create the missing issuer
+account without any flags. Instead of checking for `AUTH_CLAWBACK_ENABLED_FLAG`,
+we could just have `clawback` enabled by default, and allow the issuer to clear
+it. I don't think this is a great idea because I think we'll see most issuers
+not clear the flag even if they don't require clawback, making users hesitant to
+use classic assets on Soroban.
+
+Note that `clawback` can be disabled using `disable_clawback`, but it can not be
+enabled. This matches the clawback behavior on Stellar classic.
+
+### Balances are u128s
+
+The options considered were u63 (contained in an ScVal), u64, u128, u256, and an
+arbitrary precision big integer. Big integer is unnecessary since u256 is more
+than enough to represent any reasonable balance. This is made more obvious when
+looking at other chains and what they use (Solana uses u64, Near uses u128, and
+ERC-20 uses u256). Note that a u63 is more efficient than u64 in Soroban because
+the u63 can be contained in a 64 bit `ScVal`, while u64 will need be to an
+`ScObject`, which only exists in the host and needs to be referenced by the guest.
+
+Now the question is what should it be instead? u63 is the most efficient with
+u64 close behind, but not the most flexible if you use more than a couple
+decimal places. For example, a u64 balance with 9 decimal places would have a
+max value of 18,446,744,073.709551615. Stellar classic uses a signed 64 bit
+integer (which would be a u63 in Soroban) with seven decimal places, leading to
+a max value of 922,337,203,685.4775807. These values might be sufficient in most
+cases, but there are some edge cases where these limits are too small. For
+example, the US Treasury has a cash balance of $636 billion. This barely works
+in the Stellar case, but isn’t far off from reaching the max. It works in the
+variable decimals case if you give up some decimals. There are other currency
+and asset examples that exceed these limits as well. Instead of forcing issuers
+to compromise on decimals, the best option is to use u128 balances.
+
```

Comment:
I see that we are not providing the `decimals` function? If so, we probably need to discuss the choice.

Reply:
No this is just saying that instead of using less decimals to represent larger numbers with 64 bits, we should use 128. We have a `decimals` function in the Descriptive Interface.